### ISSUE TITLE
Is compatible with Rails 5.1.4?

### ISSUE THREAD

**Issue by pedrofurtado** [MartinKoerner/deferred_associations#2]
@MartinKoerner Is compatible with Rails 5.1.4?

**Comment by MartinKoerner** [comment]
I actually did not check that yet. I will try to ensure compatibility next weekend.

**Comment by pedrofurtado** [comment]
The only incompatibility with Rails 5.1.x is the use of ```alias_method_chain``` inside code (is deprecated).

**Comment by MartinKoerner** [comment]
@pedrofurtado actually there is another incompatibility.

In Rails 2-4, you cannot change an ID array from a HABTM association without setting a new array explicitely:

```ruby
class Course < ActiveRecord::Base
  has_and_belongs_to_many :students
end

course = Course.first
course.student_ids # we assume, it is [1]
course.student_ids << 2
course.student_ids # won't contain 2, stays at [1]
```

deferred_associations didn't change that behavior. With Rails 5.1, you can modify the array, but deferred_associations would change it back to Rails 4 behavior.

I will look into it during holidays and hopefully publish a new version with full compatibility

**Comment by MartinKoerner** [comment]
It turned out, that even Rails 5 doesn't save the IDs. So I decided to leave the functionality as it is. So with Rails 5, you get a little difference between associations with and without deferred_associations, but only for unsaved ID lists...

**Comment by pedrofurtado** [comment]
Similar solution is available: robinroestenburg/deferring#2
